node/subsystem/src/util.rs

```diff
@@ -58,11 +58,48 @@ use streamunordered::{StreamUnordered, StreamYield};
 /// Otherwise, downstream crates might have to modify their `Cargo.toml` to ensure `sp-core` appeared there.
 pub use sp_core::traits::SpawnNamed;
 
+/// Reexport prometheus types.
+pub use substrate_prometheus_endpoint as prometheus;
+
 /// Duration a job will wait after sending a stop signal before hard-aborting.
 pub const JOB_GRACEFUL_STOP_DURATION: Duration = Duration::from_secs(1);
 /// Capacity of channels to and from individual jobs
 pub const JOB_CHANNEL_CAPACITY: usize = 64;
 
+
+/// Subsystem- or job-specific prometheus metrics.
+///
+/// Usually implemented as a wrapper for `Option<ActualMetrics>`
+/// to ensure `Default` bounds or as a dummy type ().
+/// Prometheus metrics internally hold an `Arc` reference, so cloning them is fine.
+pub trait MetricsTrait: Default + Clone {
```

Comment:
Calling it `MetricsTrait` seems redundant

Reply:
It was called `Metrics` initially, but since most of struct that implement this trait were called `Metrics` (or maybe it was assoc types), I had to rename it in imports to `MetricsTrait` (rustc was not happy) and then renamed it here to match `JobTrait` naming convention. I can instead rename all the struct if that is preferred.